staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -167,6 +174,15 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr admission.Attrib
 		if callErr, ok := err.(*webhookutil.ErrCallingWebhook); ok {
 			if ignoreClientCallFailures {
 				klog.Warningf("Failed calling webhook, failing open %v: %v", hook.Name, callErr)
+
+				key := hook.Name + "." + FailureAuditAnnotationPrefix + FailureAuditAnnotationFailedOpenKeySuffix
+				value := "true"
+				if err := versionedAttr.Attributes.AddAnnotation(key, value); err != nil {
+					// In extreme rare cases, super long `hook.Name` could make the key
```

Comment:
why would we not use the same key computation as newWebhookAnnotator which handles multiple invocations, conflicting names, avoids making the webhook name part of the key and avoids length issues?

Reply:
Constructing an annotation key that simply doesn't work for some valid webhook names isn't reliable (this goes for both mutating and validation webhooks).

A webhook name must pass this validation:

* https://github.com/kubernetes/kubernetes/blob/14df10f60bbd68216f8ab9d3f30d1f938d7b659e/pkg/apis/admissionregistration/validation/validation.go#L310

* https://github.com/kubernetes/kubernetes/blob/14df10f60bbd68216f8ab9d3f30d1f938d7b659e/pkg/apis/admissionregistration/validation/validation.go#L262

* https://github.com/kubernetes/kubernetes/blob/14df10f60bbd68216f8ab9d3f30d1f938d7b659e/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L78-L90

That means a webhook name can be up to 253 characters and must pass `IsDNS1123Subdomain`.

An audit annotation key must pass this validation:

* https://github.com/kubernetes/kubernetes/blob/14df10f60bbd68216f8ab9d3f30d1f938d7b659e/staging/src/k8s.io/apiserver/pkg/admission/attributes.go#L141-L142

* https://github.com/kubernetes/kubernetes/blob/14df10f60bbd68216f8ab9d3f30d1f938d7b659e/staging/src/k8s.io/apiserver/pkg/admission/attributes.go#L202-L211

* https://github.com/kubernetes/kubernetes/blob/14df10f60bbd68216f8ab9d3f30d1f938d7b659e/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L42-L71

That means an audit annotation must have `$prefix/$suffix`, $prefix must pass `IsDNS1123Subdomain`, and $suffix cannot be longer than 63 characters.

That means the only place in an audit annotation key the webhook name can go is to be used as the entire $prefix. However:

* Using `$name/failed-open` would conflict with a `failed-open` audit annotation returned by the webhook itself
* This also claims use of a prefix not under Kubernetes' control (the point of namespacing the annotations is to avoid conflicts between uncoordinated parties).

If you want a stable audit annotation key under Kubernetes' control, something like the following would work:

* `mutating.webhook.admission.k8s.io/failed-open`
* `validating.webhook.admission.k8s.io/failed-open`

The value for those annotations could be the names of the webhooks that failed open.